### ISSUE TITLE
Carvel package support in Kubeapps

### ISSUE THREAD

**Issue by absoludity** [kubeapps/kubeapps#3221]
### Description:

We're currently working on plugin-based support for different packaging systems, moving across our existing Helm support as well as adding fluxv2 support.

We included Carvel support in the original POC, but due to it changing constantly at the time, as well as a higher priority for flux support, we paused the carvel work to focus on flux and porting the existing helm support to a plugin.

We'd love to come back and add a Carvel/kapp-controller plugin once other priorities are completed (especially as it's now stable).

**Comment by antgamdia** [comment]
Just adding a list to track the remaining operations:

- [x] List available packages
  - [x] GetAvailablePackageSummaries: https://github.com/kubeapps/kubeapps/pull/3728
  - [x] GetAvailablePackageDetail: https://github.com/kubeapps/kubeapps/pull/3735
  - [x] GetAvailablePackageVersions: https://github.com/kubeapps/kubeapps/pull/3734
- [ ] List installed packages
  - [ ] GetInstalledPackageSummaries
  - [ ] GetInstalledPackageDetail
- [ ] Action "buttons"
  - [ ] CreateInstalledPackage
  - [ ] UpdateInstalledPackage
  - [ ] DeleteInstalledPackage

Additionally, we should be fixing some minor UI issues and features we encounter (e.g., icons, filtering by package plugin, etc...) [will update this comment to track them]

**Comment by antgamdia** [comment]
I've created a bunch of PRs aimed at having initial Carvel support. However, the final result is far from being seamless and smooth. 

https://user-images.githubusercontent.com/11535726/142704788-5be9d832-52d7-4dde-ab33-ecd683c56e82.mp4

For instance, just in this demo I'd remark a couple of issues:

- The UI redirects when not every resource is created. Currently,  we are throwing errors when sth is not found. Not the best UX
- The loading times for the catalog are way too long... I'd foresee we're gonna need a cache (even if I tried using goroutines... it seems it's not enough)
  - Especially if you add the Bitnami repo as Carvel packages... 
- The demo package gets installed, but I' haven't been able to get to work another pkg but this one. I mean, not having a default values.yaml hinders a quick
